pkg/operator/osd/osd.go

```diff
@@ -299,9 +301,11 @@ func (c *Cluster) osdContainer(devices []Device, directories []Directory, select
 		VolumeMounts: volumeMounts,
 		Env:          envVars,
 		SecurityContext: &v1.SecurityContext{
-			Privileged:             &privileged,
-			RunAsUser:              &runAsUser,
-			RunAsNonRoot:           &runAsNonRoot,
+			Privileged: &privileged,
+			RunAsUser:  &runAsUser,
```

Comment:
do we need to set RunAsUser at all? whats the default?

Reply:
The default is what is set in the container image metadata.  https://kubernetes.io/docs/api-reference/v1.8/#securitycontext-v1-core

I propose to set it explicitly so kubernetes can ensure that pod is running as what we expect the image would be running as.   And by setting it explicitly now it is easier for us to review every pod type and restrict that down to the minimal needed.

Happy to remove that as well if it is redundant, but another lesson we learned is by setting the securityContext explicitly you also get more meaningful error messages at the control plane level because it mismatched.

Reply:
thats sounds reasonable. are you able to test this change on 1.6.4 and prior 1.7 onwards?

Reply:
I don't have access to a 1.6.4 cluster or 1.7.x cluster, but I can test this one a 1.8.1 cluster.